src/client.rs

```diff
@@ -226,61 +239,152 @@ impl Context {
+            SeabirdEvent::Message(message) => message
+                .source
+                .as_ref()
+                .and_then(|s| s.user.as_ref().map(|u| u.display_name.as_str())),
+            SeabirdEvent::Command(message) => message
+                .source
+                .as_ref()
+                .and_then(|s| s.user.as_ref().map(|u| u.display_name.as_str())),
+            SeabirdEvent::Mention(message) => message
+                .source
+                .as_ref()
+                .and_then(|s| s.user.as_ref().map(|u| u.display_name.as_str())),
+
+            // NOTE: PrivateMessage is in a different format
+            SeabirdEvent::PrivateMessage(message) => {
+                message.source.as_ref().map(|u| u.display_name.as_str())
+            }
         }
     }
 
     pub async fn mention_reply(&self, msg: &str) -> Result<()> {
         let sender = self
             .sender()
             .ok_or_else(|| format_err!("Tried to get the sender of an event without one"))?;
-        let target = self
-            .reply_target()
-            .ok_or_else(|| format_err!("Tried to reply to an event without a targets"))?;
 
-        // If the target matches the sender, it's a privmsg so we shouldn't send
-        // a prefix.
-        //
-        // TODO: switch to matching on PrivateMessage
-        if target == sender {
+        // If it's a private message, we shouldn't send the prefix.
+        if self.is_private() {
             self.reply(msg).await
         } else {
             self.reply(&format!("{}: {}", sender, msg)[..]).await
         }
     }
 
-    pub async fn reply(&self, msg: &str) -> Result<()> {
-        self.send_message(
-            self.reply_target()
-                .ok_or_else(|| format_err!("Tried to reply to an event without a targets"))?,
-            msg,
-        )
-        .await
-    }
-
-    pub async fn send_message(&self, target: &str, message: &str) -> Result<()> {
-        self.client.send_message(target, message).await
+    pub async fn reply(&self, text: &str) -> Result<()> {
+        match &self.raw_event {
+            SeabirdEvent::Message(message) => {
+                self.client
+                    .send_message(
+                        message
+                            .source
+                            .as_ref()
+                            .map(|s| s.channel_id.as_str())
+                            .ok_or_else(|| format_err!("message missing channel_id"))?,
+                        text,
+                    )
+                    .await
+            }
+            SeabirdEvent::Command(message) => {
+                self.client
+                    .send_message(
+                        message
+                            .source
+                            .as_ref()
+                            .map(|s| s.channel_id.as_str())
+                            .ok_or_else(|| format_err!("message missing channel_id"))?,
```

Comment:
why not just `ok_or`?

Reply:
ok_or requires you to have an owned error object, which we don't want to create every time. Eventually, it may make more sense to have a custom error type and use that though.